### PR TITLE
LNX-183 Add Take action and Agent inventory

### PR DESCRIPTION
**Pull request by EasyJB**

### CODE REVIEW

**Review by Krystian030 — APPROVED** [review]

File: lynx/common/actions/common_requirements.py [hunk 1]
```diff
@@ -61,3 +61,12 @@ def has_given_tags(scene: 'Scene', object_id: int, given_tags: List[str]) -> boo
             return True
 
         return False
+
+    @staticmethod
+    def any_object_has_given_tags_on_squere(scene: Scene, position: Vector, given_tags: List[str]) -> bool:
+        objects: List[Object] = scene.get_objects_by_position(position)
+        for object in objects:
+            if all(element in object.tags for element in given_tags):
```
Comment (filip-szweda) [inline lynx/common/actions/common_requirements.py]: `element` -> `tag`/`given_tag`

File: lynx/common/actions/common_requirements.py [hunk 2]
```diff
@@ -61,3 +61,12 @@ def has_given_tags(scene: 'Scene', object_id: int, given_tags: List[str]) -> boo
             return True
 
         return False
+
+    @staticmethod
+    def any_object_has_given_tags_on_squere(scene: Scene, position: Vector, given_tags: List[str]) -> bool:
```
Comment (filip-szweda) [inline lynx/common/actions/common_requirements.py]: `any_object_on_square_has_all_given_tags`

File: lynx/common/actions/take.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from lynx.common.actions.create_object import CreateObject
+from lynx.common.actions.remove_object import RemoveObject
+from lynx.common.object import Object
+from lynx.common.vector import Vector
+
+
+@dataclass
+class Take(Action):
+    """
+    Simple action used to hit/destroy object, which stands on the given destination.
```
Comment (filip-szweda) [inline lynx/common/actions/take.py]: Please, provide more adequate description

File: tests/common_requirements_test.py [hunk 4]
```diff
@@ -86,3 +86,17 @@ def test_existing_tree_on_square_positive(self) -> NoReturn:
 
     def test_nonexistent_object_on_square_negative(self) -> NoReturn:
         assert not CommonRequirements.is_any_on_square(self.scene, Vector(0, 0), ['test'])
+
+
+class TestIsAnyTagOnSquare:
```
Comment (filip-szweda) [inline tests/common_requirements_test.py]: Adpt to new function name

File: tests/take_test.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+from lynx.common.vector import Vector
+from lynx.common.enitity import Entity
+
+
+class TestTakeSerialization:
+    expected_serialization_take = '{"type": "Take", "attributes": {"object_id": 1, "position": {"x": 1, "y": 0}}}'
+
+    def test_success_serialization(self) -> NoReturn:
+        serialized_take = Take(object_id=1, position=Vector(1, 0)).serialize()
+
```
Comment (filip-szweda) [inline tests/take_test.py]: No need for this line, same goes below

File: tests/take_test.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        assert expected_take == deserialized_take
+
+
+class TestTakeApply:
+    def test_apply_positive(self):
+        expected_scene = Scene()
+        expected_dummy_object = Object(id=1, name="dummy", position=Vector(5, 5), inventory={"Wood": 1})
+        expected_dummy_take = Take(position=Vector(5, 6), object_id=1)
+        expected_scene.add_entity(expected_dummy_object)
+        expected_scene.add_entity(expected_dummy_take)
```
Comment (filip-szweda) [inline tests/take_test.py:31]: No need for adding `Action` entities when not comparing the whole `Scene`
Same in other test classes

File: tests/take_test.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+        dummy_take.apply(scene)
+        for action in scene.pending_actions:
+            Action.deserialize(action).apply(scene)
+        scene.pending_actions.clear()
+
+        assert expected_dummy_object == dummy_object
+        assert scene.get_objects_by_position(Vector(5, 6)) == []
+
+    def test_apply__positive(self):
```
Comment (filip-szweda) [inline tests/take_test.py]: `test_take_only_wood_from_square_with_multiple_objects_positive`

File: tests/take_test.py [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    def test_success_deserialization(self):
+        expected_take = Take(object_id=1, position=Vector(1, 0))
+
+        deserialized_take = Entity.deserialize(self.expected_serialization_take)
+
+        assert expected_take == deserialized_take
+
+
+class TestTakeApply:
+    def test_apply_positive(self):
```
Comment (filip-szweda) [inline tests/take_test.py:26]: `test_take_single_wood_from_square_positive`

File: tests/take_test.py [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        dummy_wood = Object(id=2, name="Wood", position=Vector(5, 6), tags=["pickable"])
+        dummy_tree = Object(id=3, name="Tree", position=Vector(5, 6))
+        dummy_take = Take(position=Vector(5, 6), object_id=1)
+        scene.add_entity(dummy_object)
+        scene.add_entity(dummy_wood)
+        scene.add_entity(dummy_tree)
+
+        assert dummy_take.satisfies_requirements(scene)
+
+    def test_requirements_too_far_fail(self):
```
Comment (filip-szweda) [inline tests/take_test.py]: `test_object_too_far_negative`

File: tests/take_test.py [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        dummy_wood = Object(id=2, name="Wood", position=Vector(5, 6), tags=["pickable"])
+        dummy_tree = Object(id=3, name="Tree", position=Vector(5, 6))
+        dummy_take = Take(position=Vector(5, 7), object_id=1)
+        scene.add_entity(dummy_object)
+        scene.add_entity(dummy_wood)
+        scene.add_entity(dummy_tree)
+
+        assert not dummy_take.satisfies_requirements(scene)
+
+    def test_requirements_no_pickable_fail(self):
```
Comment (filip-szweda) [inline tests/take_test.py:101]: `test_object_without_pickable_tag_negative`

File: tests/take_test.py [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        dummy_take.apply(scene)
+        for action in scene.pending_actions:
+            Action.deserialize(action).apply(scene)
+        scene.pending_actions.clear()
+
+        assert scene == expected_scene
+
+
+class TestTakeRequirements:
+    def test_requirements_positive(self):
```
Comment (filip-szweda) [inline tests/take_test.py]: `test_all_requirements_satisifed_positive`

**Review by filip-szweda — CHANGES_REQUESTED** [review]

**Review by filip-szweda — APPROVED** [review]

**Pull request merged**